src/frontend-samples/clash-review-sample/ClashReviewApp.tsx

```diff
@@ -2,170 +2,49 @@
-    const markersData: MarkerData[] = [];
-    const clashData = await ClashReviewApp.getClashData();
-
-    let count = 0;
-    for (const clash of clashData.clashDetectionResult) {
-      const point = await this.calcClashCenter(imodel, clash.elementAId, clash.elementBId);
-      const title = "Collision(s) found:";
-      const description = `Element A: ${clash.elementALabel}<br>Element B: ${clash.elementBLabel}`;
-      const clashMarkerData: MarkerData = { point, data: clash, title, description };
-      markersData.push(clashMarkerData);
-      count++;
-      if (maxClashes < count)
-        break;
-    }
-    return new Promise((resolve) => { resolve(markersData); });
-  }
-
-  private static async calcClashCenter(imodel: IModelConnection, elementAId: string, elementBId: string): Promise<Point3d> {
-    let rangeA: any;
-    const elemAProps = (await imodel.elements.getProps([elementAId])) as GeometricElement3dProps[];
-    if (elemAProps.length !== 0) {
-      elemAProps.forEach((prop: GeometricElement3dProps) => {
-        const placement = Placement3d.fromJSON(prop.placement);
-        rangeA = placement.calculateRange();
-      });
-    }
-    let rangeB: any;
-    const elemBProps = (await imodel.elements.getProps([elementBId])) as GeometricElement3dProps[];
-    if (elemBProps.length !== 0) {
-      elemBProps.forEach((prop: GeometricElement3dProps) => {
-        const placement = Placement3d.fromJSON(prop.placement);
-        rangeB = placement.calculateRange();
-      });
-    }
-
-    return rangeA.intersect(rangeB).center;
-  }
-
-  public static visualizeClashCallback = (clashData: any) => {
-    ClashReviewApp.visualizeClash(clashData.elementAId, clashData.elementBId);
+    viewState.lookAtVolume(range, aspect);
+    setViewportOptions({ viewState });
   };
 
-  public static visualizeClash(elementAId: string, elementBId: string) {
-    if (!IModelApp.viewManager.selectedView)
-      return;
-
-    const vp = IModelApp.viewManager.selectedView;
-    const emph = EmphasizeElements.getOrCreate(vp);
-    emph.clearEmphasizedElements(vp);
-    emph.clearOverriddenElements(vp);
-    emph.overrideElements(elementAId, vp, ColorDef.red, FeatureOverrideType.ColorOnly, true);
-    emph.overrideElements(elementBId, vp, ColorDef.blue, FeatureOverrideType.ColorOnly, false);
-    emph.wantEmphasis = true;
-    emph.emphasizeElements([elementAId, elementBId], vp, undefined, false);
-
-    if (applyZoom) {
-      const viewChangeOpts: ViewChangeOptions = {};
-      viewChangeOpts.animateFrustumChange = true;
-      viewChangeOpts.marginPercent = new MarginPercent(0.1, 0.1, 0.1, 0.1);
-      vp.zoomToElements([elementAId, elementBId], { ...viewChangeOpts });
-    }
-  }
-
-  public static resetDisplay() {
-    if (!IModelApp.viewManager.selectedView)
-      return;
+  return (
+    <>
+      { /* Viewport to display the iModel */}
+      {sampleIModelInfo?.contextId && sampleIModelInfo?.iModelId &&
+        <Viewer
+          productId="2686"
```

Comment:
What is productId needed for?

Reply:
If omitted from the ReadSettings sample, settings are unable to be read and an error is returned. I don't have a concrete reason why, it was there when I was porting the sample.

Can't find a reason why for the Clash-Review sample, but I have a feeling it is required for calling the API, but the api is broken right now.

Reply:
Makes sense for ReadSettings. But I doubt GPR would be needed for Clash-Review. Better to remove it.